cg/cli/balsamic.py

```diff
@@ -90,13 +93,27 @@ def config(context, dry, target_bed, case_id):
                 LOGGER.debug("destination path already exists: %s", linked_fastq_path)
 
         if link_obj.sample.is_tumour:
-            tumor_path = concatenated_paths[1]
+            tumor_paths.add(concatenated_paths[1])
         else:
-            normal_path = concatenated_paths[1]
+            normal_paths.add(concatenated_paths[1])
+
+    if not tumor_paths:
+        click.echo("No tumor sample found!", color="red")
+        context.abort()
+
+    if len(tumor_paths) > 1:
+        click.echo(f"Too many tumor samples found: {len(tumor_paths)}", color="red")
+        context.abort()
+
+    tumor_path = tumor_paths.pop()
+
+    normal_path = None
+    if len(normal_paths) > 1:
+        click.echo(f"Too many normal samples found: {len(normal_paths)}", color="red")
+        context.abort()
 
-    if not tumor_path:
-        click.echo("No tumor found!", color="red")
-        raise
+    if len(normal_paths) == 1:
+        normal_path = normal_paths.pop()
```

Comment:
Alright.

So, for tumour, we check:
- zero files
- more than one file

for normal we check
- more than one
- just one

Can you motivate that? :)

Reply:
Motivation: Balsamic is a cancer pipeline that can work with tumor only or tumor/normal as a pair and thus we need to handle the faulty inputs; 0 tumor, multiple tumor samples and multiple normals. Balsamic CAN handle multiple tumor samples but we have not implemented that call here and implementing that would be something else than a mere fix and refactoring?

Reply:
@ingkebil, happy with motivation?

Reply:
Note: Multiple tumor samples will only work with QC workflow. Calling somatic variants is not possible with multiple tumor samples.

Reply:
yes! Thanks for clarifying!

Reply:
Resolved?